### PR TITLE
[ja] Machine learning tips and tricks

### PR DESCRIPTION
**Pull request by umu1729**

### CODE REVIEW

**Comment by shervinea** [comment]
Just realized that your work was already ready to be reviewed @umu1729! Please feel free to invite friends to take a look at it.

**Comment by ytknzw** [comment]
@shervinea
This cheat sheet has been translated and reviewed in the following PR. Even though it's a different translation from umu1729's, I think we could call this cheat sheet 'done'.
https://github.com/shervinea/cheatsheet-translation/pull/164#issuecomment-510325217

**Comment by shervinea** [comment]
Hi @ytknzw, thanks for your message. It is my understanding that PR #164 is not the same as the current one, right? ([deep learning tips and tricks](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks) as opposed to [machine learning tips and tricks](https://stanford.edu/~shervine/teaching/cs-229/cheatsheet-machine-learning-tips-and-tricks))

**Comment by ytknzw** [comment]
Hi @shervinea, yes, sorry! My misunderstanding!

**Comment by shervinea** [comment]
No worries @ytknzw, thanks for helping out in seeing what can be already done! By the way, would you be interested in reviewing the translation?

**Comment by ytknzw** [comment]
> By the way, would you be interested in reviewing the translation?
Yes!

**Comment by shervinea** [comment]
@ytknzw, thanks for proposing your help! 
Hi @umu1729, please feel free to invite anyone else who you think could be interested in this process. Looking forward to the final version of this translation!

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
+**1. Machine Learning tips and tricks cheatsheet**
+
+&#10230;　機械学習チップ&トリック　チートシート
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230;機械学習のアドバイスやコツのチートシート
```
To be consistent with translation of "Deep learning tips and tricks cheatsheet":
https://stanford.edu/~shervine/l/ja/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**2. Classification metrics**
+
+&#10230;　分類評価指標
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**3. In a context of a binary classification, here are the main metrics that are important to track in order to assess the performance of the model.**
+
+&#10230; 二値分類の文脈では，次のようなモデルの性能を評価するための重要な評価指標があります．
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230;二値分類の場面では、モデルの性能を評価するために追うべき主な指標として次のものがあります。
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**4. Confusion matrix ― The confusion matrix is used to have a more complete picture when assessing the performance of a model. It is defined as follows:**
+
+&#10230; 混同行列 - 混同行列はモデルの性能を評価する際に，より完全な描像を得るために用いられます．
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; 混同行列 - 混同行列はモデルの性能を評価する際に、より完全に理解するために用いられます。次のように定義されます：
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**5. [Predicted class, Actual class]**
+
+&#10230; [予測したクラス, 実際のクラス]
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**6. Main metrics ― The following metrics are commonly used to assess the performance of classification models:**
+
+&#10230; 主要な評価指標 - 次の指標が分類モデルの性能の評価のために一般的に用いられます．
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; 主要な評価指標 - 分類モデルの性能を評価するために、一般的に次の指標が用いられます。
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**7. [Metric, Formula, Interpretation]**
+
+&#10230; [評価指標,式,解釈]
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**8. Overall performance of model**
+
+&#10230;　モデルの全体的な性能
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**9. How accurate the positive predictions are**
+
+&#10230;　正と判断された予測の正答率
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230;陽性判定の正解率
```
Or
```suggestion
&#10230;陽性的中率
```
Cf. https://ja.wikipedia.org/wiki/%E9%99%BD%E6%80%A7%E9%81%A9%E4%B8%AD%E7%8E%87

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**10. Coverage of actual positive sample**
+
+&#10230; 実際には正であるサンプルを正しく正と予測した割合
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; 本当に陽性であるサンプルを正しく判定した割合（真陽性率）
```
Cf. https://ja.wikipedia.org/wiki/%E9%99%BD%E6%80%A7%E9%81%A9%E4%B8%AD%E7%8E%87#%E5%8F%82%E8%80%83

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 11]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**11. Coverage of actual negative sample**
+
+&#10230;　実際には負であるサンプルを正しく負と予測した割合
```
Comment (ytknzw) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230;　本当に陰性であるサンプルを正しく判定した割合（偽陽性率）
```
Cf. https://ja.wikipedia.org/wiki/%E9%99%BD%E6%80%A7%E9%81%A9%E4%B8%AD%E7%8E%87#%E5%8F%82%E8%80%83

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 3 above)
Comment (tianjianjiang) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: for "の文脈では" v. "の場面では", how about "を背景に"?

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 9 above)
Comment (tianjianjiang) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: IMHO, the source phrase in English is trying to put the interpretation in layman's terms. Perhaps something similar to yet simpler than "陽性と判定された場合に、真の陽性である確率" from the aforementioned Wikipedia page.
For example, "陽性予測は、どれくらい正確ですか."

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 11 above)
Comment (scrambleegg7) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: ok

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 12]
```diff
@@ -0,0 +1,285 @@
...
+
+**44. Regression metrics**
+
+&#10230; 回帰評価指標
+
+<br>
+
+**45. [Classification metrics, confusion matrix, accuracy, precision, recall, F1 score, ROC]**
+
+&#10230; [分類評価指標,混同行列,正解率,適合率,再現率,F値,ROC]
```
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: &#10230; [分類評価指標,混同行列,正解率,適合率,再現率,F値,ROC曲線]

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 4 above)
Comment (Hiroki-Mori360) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: I suppose that `a complete picture` be translated as  '全体像' in Japanese. We use `confusion matrix` to assess the whole model, so the part in question should be like:

```
混同行列はモデルの性能を評価する際に、全体像を理解するために用いられます。
```
or something like
```
混同行列はモデルの性能を全体から評価するために用いられます。
```

The `more` in the sentence is comparing it (confusion matrix)  to other metrics which focusing on more detail part of assessment, so you can omit it in translation or a phrase '他の指標に比べてより'.

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 12 above)
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: &#10230; [分類評価指標,混同行列,正解率,適合率,再現率,F値,ROC曲線]

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 10 above)
Comment (Hiroki-Mori360) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: How about `実際に陽性であるサンプル` ?

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 11 above)
Comment (Hiroki-Mori360) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: How about `実際に陰性であるサンプル` ?

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+
+**11. Coverage of actual negative sample**
+
+&#10230;　実際には負であるサンプルを正しく負と予測した割合
+
+<br>
+
+**12. Hybrid metric useful for unbalanced classes**
+
+&#10230;　不均衡データに対する有用な複合指標
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 12 above)
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; [分類評価指標,混同行列,正解率,適合率,再現率,F値,ROC曲線]
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 14]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**13. ROC ― The receiver operating curve, also noted ROC, is the plot of TPR versus FPR by varying the threshold. These metrics are are summed up in the table below:**
+
+&#10230; ROC曲線 - 受信者動作特性曲線(ROC)は閾値を変えていく際のFPRに対するTPRのグラフです．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: Please add the next line:
```
これらの指標は下表の通りまとめられます。
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 15]
```diff
@@ -0,0 +1,285 @@
...
+<br>
+
+**12. Hybrid metric useful for unbalanced classes**
+
+&#10230;　不均衡データに対する有用な複合指標
+
+<br>
+
+**13. ROC ― The receiver operating curve, also noted ROC, is the plot of TPR versus FPR by varying the threshold. These metrics are are summed up in the table below:**
+
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: ## NOTE
An `and` should be removed from the original.

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 16]
```diff
@@ -0,0 +1,285 @@
...
+&#10230; ROC曲線 - 受信者動作特性曲線(ROC)は閾値を変えていく際のFPRに対するTPRのグラフです．
+
+<br>
+
+**14. [Metric, Formula, Equivalent]**
+
+&#10230;　[評価指標,式,等価な指標]
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: I don't know which one is the best, but I give two more examples.
- 同等の指標
- 同義の指標

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 17]
```diff
@@ -0,0 +1,285 @@
...
+
+**42. Error analysis ― Error analysis is analyzing the root cause of the difference in performance between the current and the perfect models.**
+
+&#10230; エラー分析 - エラー分析は完璧なモデルと現在のモデル間の性能差の主要な要因を分析することです．
+
+<br>
+
+**43. Ablative analysis ― Ablative analysis is analyzing the root cause of the difference in performance between the current and the baseline models.**
+
+&#10230; アブレーション分析 - アブレーション分析は，ベースラインとするモデルと現在のモデル間の性能差の主要な要因を分析することです．
```
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; アブレーション分析 - アブレーション分析は，ベースライン・モデルと改良されたモデル間で発生したパフォーマンスの差異の原因を分析することです．
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 18]
```diff
@@ -0,0 +1,285 @@
...
+
+**14. [Metric, Formula, Equivalent]**
+
+&#10230;　[評価指標,式,等価な指標]
+
+<br>
+
+**15. AUC ― The area under the receiving operating curve, also noted AUC or AUROC, is the area below the ROC as shown in the following figure:**
+
+&#10230; AUC - ROC曲線下面積(AUC,AUROC)は次の図のようにROC曲線の下側の面積のことです．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: AUC - ROC曲線下面積(AUC,AUROC)は次の図に示される通りROC曲線の下側面積のことです

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 19]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**16. [Actual, Predicted]**
+
+&#10230; [実際，予測]
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 20]
```diff
@@ -0,0 +1,285 @@
...
+
+**45. [Classification metrics, confusion matrix, accuracy, precision, recall, F1 score, ROC]**
+
+&#10230; [分類評価指標,混同行列,正解率,適合率,再現率,F値,ROC]
+
+<br>
+
+**46. [Regression metrics, R squared, Mallow's CP, AIC, BIC]**
+
+&#10230; [回帰評価指標,R二乗,マローズのCp,AIC,BIC]
```
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 21]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**47. [Model selection, cross-validation, regularization]**
+
+&#10230; [モデル選択，交差検証，正則化]
```
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; [モデルの選択，交差検証，正則化]
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 22]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**48. [Diagnostics, Bias/variance tradeoff, error/ablative analysis]**
+
+&#10230; [分析，バイアス・バリアンストレードオフ，エラー・アブレーション分析]
```
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; [解析（分析），バイアス・バリアンストレードオフ，エラー・アブレーション分析]
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 23]
```diff
@@ -0,0 +1,285 @@
...
+
+**43. Ablative analysis ― Ablative analysis is analyzing the root cause of the difference in performance between the current and the baseline models.**
+
+&#10230; アブレーション分析 - アブレーション分析は，ベースラインとするモデルと現在のモデル間の性能差の主要な要因を分析することです．
+
+<br>
+
+**44. Regression metrics**
+
+&#10230; 回帰評価指標
```
Comment (scrambleegg7) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; 回帰分析の精度評価指標
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 24]
```diff
@@ -0,0 +1,285 @@
...
+
+**16. [Actual, Predicted]**
+
+&#10230; [実際，予測]
+
+<br>
+
+**17. Basic metrics ― Given a regression model f, the following metrics are commonly used to assess the performance of the model:**
+
+&#10230;　[基本的な評価指標] 回帰モデルfが与えられたとき，次のようなよう化指標がモデルの性能を評価するために一般的に用いられます．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: &#10230;　[基本的な評価指標] 回帰モデルfが与えられたとき, モデルの性能を評価するために次のような指標が一般的に用いられます．

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 25]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**18. [Total sum of squares, Explained sum of squares, Residual sum of squares]**
+
+&#10230;　[総平方和,説明された平方和,残差平方和]
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: &#10230;　[全平方和,回帰平方和,残差平方和]

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 24 above)
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230;　[基本的な評価指標] 回帰モデルfが与えられたとき,モデルの性能を評価するために次のような指標が一般的に用いられます．
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 26]
```diff
@@ -0,0 +1,285 @@
...
+
+**31. [Generally k=5 or 10, Case p=1 is called leave-one-out]**
+
+&#10230; [k=5か10が一般的,p=1の場合はLeave-one-out cross validation法と呼ばれる．]
+
+<br>
+
+**32. The most commonly used method is called k-fold cross-validation and splits the training data into k folds to validate the model on one fold while training the model on the k−1 other folds, all of this k times. The error is then averaged over the k folds and is named cross-validation error.**
+
+&#10230;　最も一般的に用いられている方法はk交差検証法であり，データセットをk群に分け，1群を検証に，残りのk-1群を学習に用います．これをk回繰り返します．求められた検証誤差はk群全てにわたって平均化され，これは交差検証誤差と呼ばれています．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; 最も一般的に用いられている方法はk交差検証法です. データセットをk群に分けた後, 1群を検証に使用し残りのk-1群を学習に使用するという操作を順番にk回繰り返します. 求められた検証誤差はk群すべてにわたって平均化されます. この平均された誤差のことを交差検証誤差と呼びます.
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 27]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**33. Regularization ― The regularization procedure aims at avoiding the model to overfit the data and thus deals with high variance issues. The following table sums up the different types of commonly used regularization techniques:**
+
+&#10230;　正則化 - 正則化はモデルが過学習するのを避ける目的としており，したがってバリアンスが大きくなる問題に対処します．次の表は一般的に使用されるいくつかの正則化法をまとめたものです．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; 正則化 - 正則化はモデルの過学習状態を回避することが目的であり,したがってハイバリアンス問題(オーバーフィット問題)に対処できます. 一般的に使用されるいくつかの正則化法を下表にまとめました.
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 28]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**34. [Shrinks coefficients to 0, Good for variable selection, Makes coefficients smaller, Tradeoff between variable selection and small coefficients]**
+
+&#10230; [係数を0にする,変数選択に適する,係数を小さくする,変数選択と係数を小さくすることのトレードオフ]
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 29]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**35. Diagnostics**
+
+&#10230; 分析
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: I think `診断` or `診断方法` is better because the word `diagnostics` is a pair of symptoms and remedies in the later part: a metaphor.

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 30]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**36. Bias ― The bias of a model is the difference between the expected prediction and the correct model that we try to predict for given data points.**
+
+&#10230; バイアス - モデルのバイアスとは，予測するあるデータ点における，予測した結果の期待値と正しいモデルによる結果との差です．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; バイアス - モデルのバイアスとは，ある標本値群を予測する際の期待値と正しいモデルの結果との差異のことです．
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 31]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**37. Variance ― The variance of a model is the variability of the model prediction for given data points.**
+
+&#10230; バリアンス - モデルのバリアンスとは，予測するあるデータ点における，予測した結果の分散です．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; バリアンス - モデルのバリアンスとは，ある標本値群に対するモデルの予測値のばらつきのことです．
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 32]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**38. Bias/variance tradeoff ― The simpler the model, the higher the bias, and the more complex the model, the higher the variance.**
+
+&#10230; バイアス・バリアンストレードオフ - よりシンプルなモデルではバイアスが高くなり，より複雑なモデルはバリアンスが高くなります．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 33]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**39. [Symptoms, Regression illustration, classification illustration, deep learning illustration, possible remedies]**
+
+&#10230; [症状,回帰モデルでの図,分類モデルでの図,深層学習での図,可能な解決策]
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 34]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**40. [High training error, Training error close to test error, High bias, Training error slightly lower than test error, Very low training error, Training error much lower than test error, High variance]**
+
+&#10230; [高い訓練誤差,訓練誤差がテスト誤差に近い，高いバイアス,訓練誤差がテスト誤差より少しだけ小さい,極端に小さい訓練誤差,訓練誤差がテスト誤差に比べて非常に小さい,高いバリアンス]
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 35]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**41. [Complexify model, Add more features, Train longer, Perform regularization, Get more data]**
+
+&#10230; [より複雑なモデルを試す,特徴量を増やす，より長く学習する,正則化を導入する,データ数を増やす]
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]: OK

File: ja/cheatsheet-machine-learning-tips-and-tricks.md [hunk 36]
```diff
@@ -0,0 +1,285 @@
...
+
+<br>
+
+**42. Error analysis ― Error analysis is analyzing the root cause of the difference in performance between the current and the perfect models.**
+
+&#10230; エラー分析 - エラー分析は完璧なモデルと現在のモデル間の性能差の主要な要因を分析することです．
```
Comment (Hiroki-Mori360) [inline ja/cheatsheet-machine-learning-tips-and-tricks.md]:
```suggestion
&#10230; エラー分析 - エラー分析は現在のモデルと完璧なモデル間の性能差の主要な要因を分析することです．
```

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 17 above)
Comment (Hiroki-Mori360) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: I think `the current` is '現在のモデル'.

File: ja/cheatsheet-machine-learning-tips-and-tricks.md (hunk 22 above)
Comment (Hiroki-Mori360) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: I think `診断` or `診断方法` is better. We need discuss this topic.
https://github.com/shervinea/cheatsheet-translation/pull/99/files#r349865518

Comment (scrambleegg7) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: Yep, I think it might be ok with "診断", however when Google searching with combination words "機械学習" and "診断", I look at many headlines related to the medical topics in Japan. The combination words "機械学習" and ("解析" or "分析") pick up headlines related to the machine learning in Japan. Though my approaching to find best translated word not be good one, how do you think about that ?

Comment (Hiroki-Mori360) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: Yes, you're right. "診断" is a bit weird. This section is about trouble shooting when your model doesn't work as expected. But "分析" sounds like "問題分析". How about "問題分析" ?

Comment (scrambleegg7) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: Thank you for your nice comment. I agree with "問題分析". It will be definitely fit to original nuance in English.

Comment (scrambleegg7) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: Thank you for your nice comment. I agree with "問題分析". It will be definitely fit to original nuance in English.

Comment (tianjianjiang) [reply ja/cheatsheet-machine-learning-tips-and-tricks.md]: As far as I know, 打診 in Japanese had a similar word sense derivation like diagnostics in English: from medical usages to common ones.

**Comment by yoshiyukinakai** [comment]
Hello @umu1729, a team from Machine Learning Tokyo completed reviewing your translation and added some suggestions. Could you check and incorporate our suggestions?

Here is how to incorporate suggestions:
https://help.github.com/ja/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

**Comment by shervinea** [comment]
Hi @umu1729, would there be anything I could do to help out in finishing the reviewing process?

**Comment by yoshiyukinakai** [comment]
Hi @shervinea, is it possible for you to incorporate suggestions and merge? I think this translation is as good as other completed translations.

**Comment by shervinea** [comment]
Thanks @yoshiyukinakai for pinging me on this PR. I went ahead and manually merged in all relevant reviewers suggestions. Hopefully it should be good now. Thanks again to everyone who helped on this translation, really appreciate your time and work!
